src/app/shared/components/template/components/map/map.component.html

```diff
@@ -0,0 +1,27 @@
+<div class="map-container">
+  <div id="map" class="map"></div>
+  <div class="map-controls">
+    <div class="layer-control"></div>
+    @if (mapLayers.length > 0) {
+      <ion-list>
+        @for (mapLayer of mapLayers; track $index) {
+          <ion-item>
+            <ion-label>
+              <ion-toggle
+                [checked]="mapLayer.getVisible()"
+                (ionChange)="handleToggleChange($event, mapLayer)"
+              >
+                {{ mapLayer.get("name") }}
+              </ion-toggle>
+              @if (mapLayer.get("description")) {
+                <p>
+                  {{ mapLayer.get("description") }}
+                </p>
+              }
+            </ion-label>
+          </ion-item>
+        }
+      </ion-list>
+    }
+  </div>
+</div>
```

Comment:
+1 All looks really nice and concise (except the amount of boilerplate you need to add a simple toggle in ionic!).